storage/storage-client/src/lib.rs

```diff
@@ -275,6 +282,54 @@ impl StorageRead for StorageReadServiceClient {
             .boxed();
         Ok(stream)
     }
+
+    // TODO migrate this to `ConfigStorage` trait as non-async
+    // and implement for LibraDBTrait once `StorageRead` is deprecated
+    async fn batch_fetch_config(&self, access_paths: Vec<AccessPath>) -> Result<Vec<Vec<u8>>> {
+        // some access paths can have the same address, so don't duplicate requests for them
+        let addresses: Vec<AccountAddress> = access_paths
+            .iter()
+            .collect::<HashSet<_>>()
+            .into_iter()
+            .map(|path| path.address)
+            .collect();
+
+        let requests: Vec<RequestItem> = addresses
+            .iter()
+            .map(|addr| RequestItem::GetAccountState { address: *addr })
+            .collect();
+
+        let responses = self.update_to_latest_ledger(0, requests).await?.0;
+
+        // Account address --> AccountState
+        let account_states = addresses
+            .into_iter()
+            .zip(responses)
+            .map(|(addr, resp)| {
+                let account_state = AccountState::try_from(
+                    &resp
+                        .into_get_account_state_response()?
+                        .blob
+                        .ok_or_else(|| {
+                            format_err!("missing blob in account state/account does not exist")
+                        })?,
+                )?;
+                Ok((addr, account_state))
+            })
+            .collect::<Result<HashMap<_, AccountState>>>()?;
+
+        access_paths
+            .into_iter()
+            .map(|path| {
+                Ok(account_states
+                    .get(&path.address)
+                    .ok_or_else(|| format_err!("missing account state for queried access path"))?
+                    .get(&path.path)
+                    .ok_or_else(|| format_err!("no value found in account state"))?
+                    .clone())
+            })
+            .collect()
```

Comment:
doest this work? it looks like a Vec<Result<_>> instead of a Result<Vec<_>>

Reply:
seems like it does-  also otherwise compiler would have complained?
https://stackoverflow.com/questions/26368288/how-do-i-stop-iteration-and-return-an-error-when-iteratormap-returns-a-result